### PR TITLE
add relevant roles for power consumption and delivery

### PR DESCRIPTION
**Pull request by DutchmanNL**
* value.power.current       (unit=W or KW)
* value.power.delivery      (unit=Wh or KWh)

### CODE REVIEW

File: doc/STATE_ROLES.md [hunk 1]
```diff
@@ -68,7 +68,9 @@
 * value.lock            - actual position of lock
 * value.max
 * value.min
-* value.power.consumption (unit=Wh or KWh)
+* value.power.consumption   (unit=Wh or KWh)
```
Comment (AlCalzone) [inline doc/STATE_ROLES.md]: Nitpicking, but the correct unit would be `kWh`, not `KWh` :grimacing:

Comment (DutchmanNL) [reply doc/STATE_ROLES.md]: yeah aggree, fixed

**Comment by Apollon77** [comment]
Is "delivery" the correct term?
Also "current" is reserved word in area of power and powe.current may be wrong interpreted.

Formally correct:
* Momentanwert: instantaneous
* Lieferung Strom: power supply

Sag doch mal die Deutschen Worte ...

**Comment by DutchmanNL** [comment]
> Sag doch mal die Deutschen Worte ...

sagst du nem Hoellaender :P 

Im Discovergy adapter (und deren seite) nennen wir die Sachen eigentlich : 
- Abgabe
- Abnahme oder Verbrauch

Power supply wird nicht benutzt in Kombination mit Lieferung sonder eher ein gerät das ström bereit stellt (eine battery usw).

**Comment by AlCalzone** [comment]
> delivery

I'd say production and consumption.

**Comment by Jey-Cee** [comment]
Please check dictindustry.com

**Comment by AlCalzone** [comment]
**Instantaneous** power is correct for "the current/actual power" as far as I can find. I don't like using "actual" for that, as it mostly has a different meaning than in german:

> **actual**
> **tatsächlich**, aktuell, wirklich, eigentlich, effektiv, konkret

I don't know about **output**. IMO it doesn't really capture the intent of the amount of energy that was **fed** or **injected** into the grid.

**Comment by DutchmanNL** [comment]
for me output has nothing to do with energy ;)

but basically for current its just power, meaning the current used W. or kW.
But still here we have consumption, delivery AND production 
(you consume from the net, you deliver to the net and you Produce by dollar)

all has todo with power.
than we have history (kWh / Wh) a example, so by making the choice for the power part we also do it for the past...

so example :

value.power.delivered ==> history delivered in Wh/kWh
value.power.consumed ==> history consumed in Wh/kWh
value.power.delivery ==> current delivered power
value.power.consumption ==> current delivered power

we could also keep the last to as is, but then we do not separate the states but its a quite common use. case

**Comment by Apollon77** [comment]
Delivery has multi meaning because it do not say if YOU deliver it to the outside (selling) or if you GET it delivered FROM the outside (purchasing). We should be clear there. 

* generating or production is „what you roof produces“

Maybe then really incoming and outgoing or selling and buying?

**Comment by DutchmanNL** [comment]
> Maybe then really incoming and outgoing or selling and buying?

interesting thought as production is indeed what you produce and not what you deliver back.
maybe contribution is the better wording here ?

**Comment by AlCalzone** [comment]
> value.power.delivered ==> history delivered in Wh/kWh
> value.power.consumed ==> history consumed in Wh/kWh

That is not power, that is energy. Power is energy/time, so the accumulation (or integral) of that is just energy. 🤓

**Comment by Apollon77** [comment]
Ok guys ,... how to continue here?

**Comment by AlCalzone** [comment]
I just worked on the Meter CC in Z-Wave. The [specs](https://github.com/AlCalzone/node-zwave-js/blob/master/specs/SDS13781-Z-Wave-Application-Command-Class-Specification.pdf) (page 326) define the stuff as follows, which makes sense IMO:

A meter can return **instantaneous** or **accumulated** readings. In our case that would be current power (instantaneous) or energy (accumulated).
A meter can differentiate between **consumed** resources and **produced** resources.
Furthermore, there are 5 defined meter types: Electric, Gas, Water, Heating and Cooling.

I think we treat this as prior art and do something similar.

**Comment by GermanBluefox** [comment]
Anyway it is wrong repo. This is valid place https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md

**Comment by Apollon77** [comment]
Heißt das das die GitHub Wiki Seite mir den Rollen outdated ist? Dann bitte löschen und linken !!

**Pull request closed without merging**